### ISSUE TITLE
Datepicker is not accessible

### ISSUE THREAD

**Issue by annam002** [buefy/buefy#2136]
### Overview of the problem
The datepicker component is not accessible for keyboard and screenreader users
**Buefy** version: 0.8.9
**Vuejs** version: 2.6.10
**OS/Browser**: Chrome

### Description
The datepicker component is not accessible for keyboard and screenreader users. We would like to use it, but the non-accessibility makes it a no-go.
Some of the problems I found:
- Focus does not jump into the datepicker field when opening the datepicker
- The tab order of the date picker header is wrong
- Selecting a day is done with tabs which is quite a hassle. It's easy to tab out of the datepicker by accident. The arrow keys should be used instead
- For screen reader users, the structure of the datepicker is completely unclear, because there is no semantic html used, just divs.
- The arrow buttons do not have labels, so a screen reader user does not know what they do
- The date input field is readonly. For the keyboard and screenreader users this would be the fastest way to fill in the date

### Steps to reproduce
- Try to use the datepicker without a mouse or with a screen reader (close eyes for the full experience)

### Expected result
An accessible datepicker. There is an example in the Wai-Aria authorizing practices: https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/datepicker-dialog.html

**Comment by service-paradis** [comment]
Thanks for this complete feedback about Datepicker a11y!
There is a solution for the last bullet: a prop called `editable`.

**Comment by jtommy** [comment]
@annam002 thanks a lot for the review, you're right.
About first point there is `focusable` prop and you might set it to false but in the next breaking I want to set it to false as default.

**Comment by annam002** [comment]
Thanks for the quick replies. Indeed, I overlooked the editable property. Nice!
The correct tabbing order for the header would be the order in which the elements are displayed. 
I'm not sure what the "focusable" prop does. I set it to false but did not see any effect.

**Comment by jtommy** [comment]
@annam002 using tab the focus is on first element of datepicker

**Comment by service-paradis** [comment]
@annam002 
You're right about the tab order. It is related to how Bulma Pagination works. Related to https://github.com/buefy/buefy/issues/1461#issuecomment-500617549.

**Comment by annam002** [comment]
@jtommy yes, with tab I can tab into the datepicker. But that works regardless of the focusable prop. At least in Chrome, didn't test it on other browsers.
@service-paradis I see. So, when I zoom in, the order of the elements in the header changes and the tab order is right. Maybe you could consider not changing the order on zooming. Or is there a really good reason why it should be different in this case?

**Comment by service-paradis** [comment]
Juste created a new PR (#2207) with some improvements.
Would love to have your feedback on it @annam002!

**Comment by jtommy** [comment]
@annam002 I've just merged #2207, what do you think ?

**Comment by RiqueBR** [comment]
Hi guys, amazing work here. These changes are working on chrome but not on firefox, chrome on IE. Any chance these changes could be across different browsers?

**Comment by jtommy** [comment]
What do you mean?

**Comment by RiqueBR** [comment]
On chrome I am able to: 
- Tab and press enter to open datepicker, 
- Tab between prev/next arrows,
- Tab between month/year dropdown and use arrow keys to navigate and select something, 
- I can tab to the dates and navigate around using arrow buttons and select a date.

However, on firefox, Safari and IE I'm only able to: 
- Tab and open datepicker pressing enter
- Tab between month/year dropdows

To explain it further, I can't tab to prev/next arrows, I also can't tab to a date and navigate with arrow keys. I can't navigate with the tab button either. 

Hope it makes sense :)

**Comment by service-paradis** [comment]
@RiqueBR 
Working fine on Firefox 77.0.1 (64 bits) and Firefox 78.0.1 (64 bits) on Windows 10.
Same for Microsoft Edge 44.19041.1.0

I was able to reproduce the issue with IE11 though.

**Comment by service-paradis** [comment]
I know what is going on! https://developer.mozilla.org/fr/docs/Web/API/KeyboardEvent/key/Key_Values#Navigation_keys
> Internet Explorer, Edge (16 and earlier), and Firefox (36 and earlier) use "Left", "Right", "Up", and "Down" instead of "ArrowLeft", "ArrowRight", "ArrowUp", and "ArrowDown".

**Comment by RiqueBR** [comment]
Hey @service-paradis, thank you so much for checking this out. I'll attach my wee implementation here, am I missing something because I have version 77.0.1 (x64) and it's still an issue for me. Bear in mind I'm on a mac. How can I implement it then?

```
<b-field label="Select a date">
        <b-datepicker
          placeholder="Type or select a date..."
          icon="calendar-today"
          focusable="true"
          v-model="today"
          trap-focus
          editable>
        </b-datepicker>
    </b-field>
```

**Comment by service-paradis** [comment]
I will send a new PR that will use older keys. I hope it will fix it (cannot confirm on a mac right now).
